### PR TITLE
[DEV APPROVED] 8974 & 8798 - Remove duplicate views on land transaction tax and land building transaction tax

### PR DESCRIPTION
**Pull request by tomas-stefano**
[TP Card](https://moneyadviceservice.tpondemand.com/entity/8798-land-and-buildings-transaction-tax-scotland)
[TP Card](https://moneyadviceservice.tpondemand.com/entity/8974-land-transaction-tax-calculator-stamp-duty)

## Context

The views were duplicated for the lbtt for Scotland and
ltt for Wales, so I decided to use Rails feature "Single template inheritance" to remove the duplication.
I organised the controller and views in the following manner:

Each calculator controller will have two methods:

* I18n key - corresponding to the key to look on the translations for each calculator.
* The form path - each calculator has your own submit form url.

And the **show** view is organised as:
```
- app/views/mortgage_calculator/transaction_tax_calculator (the parent of both controllers)
   - show.html.erb
      - calculator_form
      - links_to_other_tools
      - results_recalculation_for
```
With this it is possible to remove all duplicated views from those two calculators.

The only template that is different between the calculators are the links to other tools which is separated using single template inheritance as well:
```
├── land_and_buildings_transaction_taxes
│   └── _links_to_other_tools.html.erb
├── land_transaction_taxes
│   └── _links_to_other_tools.html.erb
```
## What this PR did not address and will not address

**This PR focus only on remove the duplication of the views and does not and will not refactor any view.**

### CODE REVIEW

File: app/views/mortgage_calculator/land_and_buildings_transaction_taxes/_bands_table.html.erb [hunk 1]
```diff
@@ -1,18 +0,0 @@
-<table class="mortgagecalc__table stamp-duty__table">
-  <thead>
-    <tr>
-      <th class="mortgagecalc__table__price"><%= I18n.t("land_and_buildings_transaction_tax.table.property_price_header") %></th>
```
Comment (willhall88) [inline app/views/mortgage_calculator/land_and_buildings_transaction_taxes/_bands_table.html.erb]: Given you have now refactored this here
https://github.com/moneyadviceservice/mortgage_calculator/pull/333/files#diff-c03244dd20a04b073f19ce0ad3d866b7R7

you can now remove the `land_and_buildings_transaction_tax.table` translations

File: app/views/mortgage_calculator/transaction_tax_calculator/_next_home_buyer_explanation.html.erb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<p><%= I18n.t("land_transaction_tax.how_calculated") %></p>
+<p><%= I18n.t("land_transaction_tax.how_calculated_additional") %></p>
+
+<table class="mortgagecalc__table stamp-duty__table">
+  <thead>
+    <tr>
+      <th class="mortgagecalc__table__price">
+        <%= I18n.t("land_transaction_tax.table.property_price_header") %>
```
Comment (willhall88) [inline app/views/mortgage_calculator/transaction_tax_calculator/_next_home_buyer_explanation.html.erb]: either that or you need to refactor this to 
`<%= I18n.t("#{i18n_locale_namespace}.table.property_price_header") %>`

Comment (tomas-stefano) [reply app/views/mortgage_calculator/transaction_tax_calculator/_next_home_buyer_explanation.html.erb]: Good spot. I already fixed ♻️. After this fix I run the commands: `grep -r "I18n.t('land_and" app/views` and `grep -r "I18n.t('land_tra" app/views` and they didn't return anything, so I think now is clear.

Comment (willhall88) [reply app/views/mortgage_calculator/transaction_tax_calculator/_next_home_buyer_explanation.html.erb]: nice! and both `land_transaction_tax` and `land_and...` have all the translation keys that `I18n.t("#{i18n_locale_namespace}` applies to them?

Comment (tomas-stefano) [reply app/views/mortgage_calculator/transaction_tax_calculator/_next_home_buyer_explanation.html.erb]: Yes, they do. And the dummy app is setup to raise errors if locale does not exist.

**Pull request merged**